examples/set-label/functions/fn-config.yaml

```diff
@@ -0,0 +1,19 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: my-config
+  annotations:
+    config.k8s.io/function: |
+      container:
+        image: gcr.io/kpt-functions/set-label
+    config.kubernetes.io/local-config: "true"
+data:
+  labels:
+  - label_name: color
+    label_value: orange
+  - fieldSpecs:
```

Comment:
@droot We need to rationalize these with `filters` at the Pipeline level.

@Shell32-Natsu @mengqiy For context, instead of each function implementing this filtering mechanism, we can provide this at a higher level in the `Pipeline` which makes function implementation simpler.

Reply:
After looking at this and offline discussions, here is the summary:
`fieldSpecs` provided here is the schema information that `set-label` needs to support different types. So this completely orthogonal to the `filtering` mechanism in the pipeline.

Another bit that confused me is that currently schema and input args (label/value in this case) is mixed together that makes the usage look very different from the case where schema is not provided, so there is an opportunity to unify that as well. 

so, TL;DR is we need a mechanism to make schema information available to all KRM functions in an easy way. And while doing so, we should also unify the UX for taking input for the functions with or without schema info.

Reply:
To expand on this, I consider the `fieldSpec` to be a low-level crutch until we address the general problem of augmenting the OpenAPI document with such semantics. A general solution would address all the following:

- Which field in a type is a `label`?
- Where is a field references from other resources?
- etc.

Kustomize has hardcoded these for k8s native types, we need to generalize this to address other resources types, e.g. Cloud Connector types.